src/main/kotlin/org/rust/ide/inspections/checkMatch/Pattern.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.inspections.checkMatch
+
+import org.rust.lang.core.psi.RsEnumVariant
+import org.rust.lang.core.psi.RsStructItem
+import org.rust.lang.core.psi.ext.RsElement
+import org.rust.lang.core.psi.ext.RsFieldsOwner
+import org.rust.lang.core.resolve.VALUES
+import org.rust.lang.core.resolve.processNestedScopesUpwards
+import org.rust.lang.core.types.ty.Ty
+import org.rust.lang.core.types.ty.TyAdt
+import org.rust.lang.core.types.ty.TyTuple
+import org.rust.lang.core.types.ty.TyUnknown
+
+data class Pattern(val ty: Ty, val kind: PatternKind) {
+    private fun RsElement.findInScope(name: String): Boolean =
+        processNestedScopesUpwards(this, VALUES) { entry -> entry.name == name }
+
+    fun text(ctx: RsElement? = null): String =
+        when (kind) {
+            is PatternKind.Wild -> "_"
+
+            is PatternKind.Binding -> kind.name
+
+            is PatternKind.Variant -> {
+                val variantName = kind.variant.name.orEmpty()
+                val itemName = kind.item.name
+
+                val isVariantInScope = ctx?.findInScope(variantName) ?: false
+                val name = if (isVariantInScope) variantName else "$itemName::$variantName"
```

Comment:
We should check that the variant we resolved is equals to `kind.item`. And let's add TODO about autoimport

Reply:
@vlad20012  I think it should be an issue, not just TODO